dialogue-apache-hc4-client/src/main/java/com/palantir/dialogue/hc4/ApacheHttpClientChannels.java

```diff
@@ -410,10 +433,31 @@ public CloseableClient build() {
             });
 
             CloseableHttpClient apacheClient = builder.build();
-            return CloseableClient.wrap(apacheClient, name, connectionManager, conf, executor);
+            IdleConnectionEvictor connectionEvictor = new IdleConnectionEvictor(
+                    connectionManager,
+                    idleConnectionEvictorThreadFactory(name, conf.taggedMetricRegistry()),
+                    // Use a shorter check duration than idle connection timeout duration in order to avoid allowing
+                    // stale connections to race the server-side timeout.
+                    Math.min(idleConnectionTimeoutMillis, 5_000),
```

Comment:
Basically 5 seconds with the caveat that a very, very low idle timeout will result in more frequent sweeps.